src/Interactive/Host/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -367,44 +358,42 @@ private static string GenerateUniqueChannelLocalName()
             /// Reads given initialization file (.rsp) and loads and executes all assembly references and files, respectively specified in it.
             /// Execution is performed on the UI thread.
             /// </summary>
-            [OneWay]
-            public void InitializeContext(RemoteAsyncOperation<RemoteExecutionResult> operation, string initializationFile, bool isRestarting)
+            public void InitializeContext(string initializationFile, bool isRestarting)
             {
-                Debug.Assert(operation != null);
 
+                var completionSource = new TaskCompletionSource<RemoteExecutionResult>();
                 lock (_lastTaskGuard)
                 {
-                    _lastTask = InitializeContextAsync(_lastTask, operation, initializationFile, isRestarting);
+                    _lastTask = InitializeContextAsync(_lastTask, completionSource, initializationFile, isRestarting);
                 }
             }
 
             /// <summary>
             /// Adds an assembly reference to the current session.
             /// </summary>
-            [OneWay]
-            public void AddReference(RemoteAsyncOperation<bool> operation, string reference)
+            public void AddReference(string reference)
```

Comment:
>AddReference [](start = 24, length = 12)

This will need to be `async Task AddReference` (all remote APIs need to be async now).